### DISCUSSION TITLE
Open a dialog with a method call

### DISCUSSION THREAD

**Discussion by maku** [spartan-ng/spartan#168] [Q&A]
I would be interested in how to open a spartan dialog with code and not include the code in the template. e.g. usage like @angular/cdk/Dialog -> dialog.open(Component).

Is there a way to do this?

**Comment by sheikalthaf** [comment]
@maku Dynamic Component in Dialog is implemented in Dialog but not yet released #180

**Reply by maku** [reply]
Thank you

**Reply by goetzrobin** [reply]
This code is now released. You can also use a signal passed to the dialog component to drive the state from your  TypeScript class!
